site-src/geps/gep-1058.md

```diff
@@ -0,0 +1,962 @@
+# GEP-1058: Route Inclusion and Delegation
+
+- Issue: [#1058](https://github.com/kubernetes-sigs/gateway-api/issues/1058)
+- Status: Proposed
+
+## TLDR
+
+Add support for a 2-tier hierarchy of route configuration to allow partial
+delegation of route settings and, provide more flexibility in how route
+configurations are organized and managed.
+
+## Goals
+
+- Provide a 2-tier hierarchy of route configuration to allow partial delegation
+of route settings.
+- Allow a route owner to limit the scope of a specific delegation.
+- Provide a method to reuse a subset of route configuration across multiple routes
+- Minimize the changes to the existing Gateway API
+- Focus on httpRoute but create a design pattern that can be applied to other
+route types in the future
+
+## Non-Goals
+
+- Explicitly define how route inclusion/delegation works for any specific route
+  type other than `HTTPRoute`
+
+## Terminology
+
+- **Route**: A generic substitute for the type of route. Equivalent to saying,
+“a route of type [*type*]”
+- **Parent Route:** A `Route` that specifies a listener as its parentRef object
+and is listed as a parentRef object in a different Route. Technically, a parent
+route is any route that has a non-nil AllowedRoutes block
+- **Child Route:** A Route that lists a Route as its parentRef object.
+- **Ephemeral Route:** The route configuration that results from combining (i.e.
+ reconciling) a parent route and the child route(s) that attaches to it. The
+ ephemeral route represents the configuration that is actually applied to the
+ data plane.
+- **Delegation:** when an included route crosses a persona ownership boundary
+(e.g. a parent route is managed by an infrastructure team and a child route is
+managed by an app dev team)
+- **Inclusion:** When a child Route’s configuration is combined with a parent
+Route’s config during runtime. The two routes may or may not be owned by the same
+persona or organization. All delegations are inclusions, but not all inclusions
```

Comment:
Also defining requirements for both inclusion and delegation, and maybe having separate docs since I don't think one solution can satisfy both.

I don't think it is safe for inclusion to cross namespace boundaries - while delegation main purpose is for crossing. 
The biggest problem in this spec is mixing the 2 and failing to address either problem in a simple way.